OpenRA.Mods.Common/Traits/Crates/LevelUpCrateAction.cs

```diff
@@ -67,12 +67,12 @@ public override void Activate(Actor collector)
 			if (info.MaxExtraCollectors > -1)
 				inRange = inRange.Take(info.MaxExtraCollectors);
 
-			foreach (var actor in inRange.Append(collector))
+			foreach (var recipient in inRange.Append(collector))
 			{
-				var recipient = actor;	// loop variable in closure hazard
 				recipient.World.AddFrameEndTask(w =>
 				{
-					recipient.TraitOrDefault<GainsExperience>()?.GiveLevels(info.Levels);
+					if (!recipient.IsDead)
+						recipient.Trait<GainsExperience>().GiveLevels(info.Levels);
```

Comment:
Won't this now crash if actors without `GainsExperience` pick up a level-up crate, though?

Reply:
Oh right, the loop above only checks if `inRange` has the correct trait, not if `collector` has it, good spotting.